moviebot/explainability/explainable_user_model.py

```diff
@@ -0,0 +1,30 @@
+"""Abstract class for creating explainable user models."""
+
+from abc import ABC, abstractmethod
+from typing import Dict
+
+from dialoguekit.core import AnnotatedUtterance
+
+UserPreferences = Dict[str, Dict[str, float]]
+
+
+class ExplainableUserModel(ABC):
+    @abstractmethod
+    def generate_explanation(
+        self, user_preferences: UserPreferences
+    ) -> AnnotatedUtterance:
+        """Generate an explanation based on the provided input data.
+
+        Args:
+            input_data: The input data for which an explanation is to be
+            generated.
+
+        Returns:
+            The generated explanation.
```

Comment:
Suggestion: Maybe this could be more precise, like utterance containing explanation.

Reply:
Done